stdlib/public/stubs/Stubs.cpp

```diff
@@ -249,13 +255,60 @@ extern "C" uint64_t swift_float80ToString(char *Buffer, size_t BufferLength,
 }
 
 /// \param[out] LinePtr Replaced with the pointer to the malloc()-allocated
-/// line.  Can be NULL if no characters were read.
+/// line.  Can be NULL if no characters were read. This buffer should be
+/// freed by the caller if this function returns a positive value.
 ///
 /// \returns Size of character data returned in \c LinePtr, or -1
 /// if an error occurred, or EOF was reached.
 ssize_t swift::swift_stdlib_readLine_stdin(char **LinePtr) {
+#if defined(_MSC_VER)
+  if (LinePtr == nullptr)
+    return -1;
+
+  int Capacity = 0;
+  ssize_t Pos = 0;
+  char *ReadBuf = nullptr;
+
+  _lock_file(stdin);
+
+  for (;;) {
+    int ch = _fgetc_nolock(stdin);
+
+    if (ferror(stdin) || (ch == EOF && Pos == 0)) {
+      if (ReadBuf)
+        free(ReadBuf);
+      _unlock_file(stdin);
+      return -1;
+    }
+
+    if (Capacity - Pos <= 1) {
+      // Capacity changes to 128, 128*2, 128*4, 128*8, ...
+      Capacity = Capacity ? Capacity * 2 : 128;
+      char *NextReadBuf = static_cast<char *>(realloc(ReadBuf, Capacity));
+      if (NextReadBuf == nullptr) {
+        if (ReadBuf)
+          free(ReadBuf);
+        _unlock_file(stdin);
+        return -1;
+      }
+      ReadBuf = NextReadBuf;
+    }
+
+    if (ch == EOF)
+      break;
+    ReadBuf[Pos++] = ch;
+    if (ch == '\n')
+      break;
+  }
+
+  ReadBuf[Pos] = '\0';
```

Comment:
Wouldn't this overflow the buffer in case `Capacity - Pos` was 2 in the check above?